File: application/src/main/java/com/highpowerbear/hpbanalytics/rest/model/CalculateStatisticsRequest.java
package com.highpowerbear.hpbanalytics.rest.model;

import java.time.temporal.ChronoUnit;

/**
 * Created by robertk on 8/25/2020.
 */
public class CalculateStatisticsRequest {

    private ChronoUnit interval;
    private String tradeType;
    private String secType;
    private String currency;
    private String underlying;

    public ChronoUnit getInterval() {
        return interval;
    }

    public String getTradeType() {
        return tradeType;
    }

    public String getSecType() {
        return secType;
    }

    public String getCurrency() {
        return currency;
    }

    public String getUnderlying() {
        return underlying;
    }
}

File: application/src/main/java/com/highpowerbear/hpbanalytics/rest/model/CalculateStatisticsRequest.kt
package com.highpowerbear.hpbanalytics.rest.model

import java.time.temporal.ChronoUnit

/**
 * Created by robertk on 8/25/2020.
 */
class CalculateStatisticsRequest {
    val interval: ChronoUnit? = null
    val tradeType: String? = null
    val secType: String? = null
    val currency: String? = null
    val underlying: String? = null
}